extensions/wikia/AjaxPoll/AjaxPoll_body.php

```diff
@@ -149,54 +147,48 @@ public function getVotes($isSubmit = false) {
 		$votes = array();
 		$total = 0;
 
-		$memcKey = wfMemcKey(self::MEMC_PREFIX_GETVOTES, $this->mId);
-		if (!$isSubmit) {
-			$votes = $wgMemc->get($memcKey);
+		$memcKey = $this->getVotesMemKey();
+		if ( !$isSubmit ) {
+			$votes = $wgMemc->get( $memcKey );
 		}
-		if (!empty($votes)) {
-			foreach($votes as $ansData){
+		if ( !empty( $votes ) ) {
+			foreach( $votes as $ansData ){
 				$total += $ansData["value"];
 			}
 		} else {
-			$votes = array();
 			$dbr = wfGetDB( $isSubmit ? DB_MASTER : DB_SLAVE );
-			$oRes = $dbr->select(
-				array( "poll_vote" ),
-				array( "poll_answer" ),
-				array( "poll_id" => $this->mId ),
-				__METHOD__
-			);
 
-			while( $oRow = $dbr->fetchObject( $oRes ) ) {
-				if( isset( $votes[ $oRow->poll_answer ] ) ) {
-					$votes[ $oRow->poll_answer ][ "value" ]++;
-				} else {
-					$votes[ $oRow->poll_answer ][ "value" ] = 1;
-				}
-				$total++;
-			}
-			$dbr->freeResult( $oRes );
+			$votes = ( new WikiaSQL() )
+				->SELECT( 'poll_answer' )
+					->COUNT( '*' )->AS_( 'count' )
+				->FROM( 'poll_vote' )
+				->WHERE( 'poll_id' )->EQUAL_TO( $this->mId )
+				->GROUP_BY( 'poll_answer' )
+				->runLoop( $dbr, function( &$votes, $row ) use ( &$total ) {
+					$votes[$row->poll_answer]['value'] = $row->count;
+					$total += $row->count;
+				}, [] );
```

Comment:
I don't believe you need to pass this empty array as the third parameter. That's the default value in the function signature.